### PR TITLE
feat(design): add accessibility considerations to the notification component

### LINKED ISSUE #2891: [FEAT] Accessibility Considerations for DaffNotificationComponent
# :bulb: Feature request

## Feature Name
Accessibility Considerations for DaffNotificationComponent

## The Desired Behavior
Notifications with actions or have a status of `warn` of `danger` should have `role="alert"`
Notifications without actions and is a informational (default) or has a status of `success` should have `role="status"`

## Prior Work
Better accessibility considerations for the notification component

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Notification's accessibility considerations are not comprehensive enough.

Fixes: #2891 

## What is the new behavior?
1. `warn` and `danger` statuses have a `role="alert"`
2. notifications with actions also have a `role="alert"`
3. all other notifications (i.e. default (no status), `success` status) have a `role="status"`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/design/notification/README.md [hunk 1]
```diff
@@ -44,4 +46,4 @@ The close button is hidden by default but can be visible by setting the `dismiss
 <design-land-example-viewer-container example="dismissible-notification"></design-land-example-viewer-container>
 
 ## Accessibility
-Notifications sets `aria-live` to `polite` by default. It does not interrupt a user's current activity and waits until they are idle to make the announcement.
\ No newline at end of file
+Notifications with actions or with a `danger` or `warn` status have a `role="alert"` so that it can be announced by assistive technologies. All other notifications have a `role="status"`. Notifications have a `tabindex="0"` so users can discover them while tabbing through a page.
```
Comment (damienwebdev) [inline libs/design/notification/README.md]: Link to the MDN docs

File: libs/design/notification/src/notification/notification.component.ts [hunk 2]
```diff
@@ -88,12 +96,14 @@ export class DaffNotificationComponent
...
+  constructor(
+    private elementRef: ElementRef,
+    private renderer: Renderer2,
+  ) {
 	  super(elementRef, renderer);
   }
 
-  @ContentChild(DaffPrefixDirective)
-    _prefix: DaffPrefixDirective;
+  @Output() closeNotification: EventEmitter<void> = new EventEmitter();
```
Comment (damienwebdev) [inline libs/design/notification/src/notification/notification.component.ts:113]: Needs docs.

File: libs/design/notification/src/notification/notification.component.ts [hunk 3]
```diff
@@ -55,10 +58,17 @@ export class DaffNotificationComponent
...
+  @ContentChild(DaffPrefixDirective) _prefix: DaffPrefixDirective;
+
+  @ContentChild(DaffNotificationActionsDirective) _actions: DaffNotificationActionsDirective;
+
   @HostBinding('class.daff-notification') class = true;
 
   @HostBinding('attr.tabindex') tabindex = '0';
-  @HostBinding('attr.aria-live') ariaLive = 'polite';
+
+  @HostBinding('attr.role') get role() {
```
Comment (damienwebdev) [inline libs/design/notification/src/notification/notification.component.ts:73]: needs docs.

File: libs/design/notification/src/notification/notification.component.ts [hunk 4]
```diff
@@ -55,10 +58,17 @@ export class DaffNotificationComponent
   implements DaffPrefixable, DaffStatusable {
   faTimes = faTimes;
 
+  @ContentChild(DaffPrefixDirective) _prefix: DaffPrefixDirective;
```
Comment (damienwebdev) [inline libs/design/notification/src/notification/notification.component.ts:61]: This should be `private`

File: libs/design/notification/src/notification/notification.component.ts [hunk 5]
```diff
@@ -55,10 +58,17 @@ export class DaffNotificationComponent
...
+
+  @ContentChild(DaffNotificationActionsDirective) _actions: DaffNotificationActionsDirective;
```
Comment (damienwebdev) [inline libs/design/notification/src/notification/notification.component.ts:63]: This should be `private`

File: libs/design/notification/src/notification/notification.component.ts [hunk 6]
```diff
@@ -55,10 +58,17 @@ export class DaffNotificationComponent
...
+
   @HostBinding('class.daff-notification') class = true;
 
   @HostBinding('attr.tabindex') tabindex = '0';
-  @HostBinding('attr.aria-live') ariaLive = 'polite';
+
+  @HostBinding('attr.role') get role() {
+    return this._actions || this.status === DaffStatusEnum.Warn || this.status === DaffStatusEnum.Danger ? 'alert' : 'status';
```
Comment (damienwebdev) [inline libs/design/notification/src/notification/notification.component.ts]: Should `info` notifications with actions have status `alert`? I think this should just follow from the `status` prop.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/design/notification/src/notification/notification.component.ts (hunk 4 above)
Comment (xelaint) [reply libs/design/notification/src/notification/notification.component.ts:61]: can't be private

File: libs/design/notification/src/notification/notification.component.ts (hunk 5 above)
Comment (damienwebdev) [reply libs/design/notification/src/notification/notification.component.ts:63]: Woops. Looks like the template uses this and can't be private.

Comment (xelaint) [reply libs/design/notification/src/notification/notification.component.ts:63]: can't be private

File: libs/design/notification/src/notification/notification.component.ts (hunk 4 above)
Comment (damienwebdev) [reply libs/design/notification/src/notification/notification.component.ts:61]: Woops. Looks like the template uses this and can't be private.

File: libs/design/notification/src/notification/notification.component.ts (hunk 6 above)
Comment (xelaint) [reply libs/design/notification/src/notification/notification.component.ts]: updated

File: libs/design/notification/src/notification/notification.component.ts (hunk 3 above)
Comment (xelaint) [reply libs/design/notification/src/notification/notification.component.ts:73]: added

File: libs/design/notification/src/notification/notification.component.ts (hunk 2 above)
Comment (xelaint) [reply libs/design/notification/src/notification/notification.component.ts:113]: added

File: libs/design/notification/README.md (hunk 1 above)
Comment (xelaint) [reply libs/design/notification/README.md]: added

**Review by damienwebdev — APPROVED** [review]

**Pull request merged**